Migrate to GTFS-rt predictions and locations feeds

PaulJKim opened the pull request:
#### Summary of changes
**Asana Ticket:** [Rearchitect RTS off of RTR](https://app.asana.com/0/1185117109217413/1204787443634599/f)

This change adjusts the predictions parsing logic to allow RTS to use the GTFS-rt feeds instead of the RTR feeds. The locations parsing logic doesn't need to be change because the GTFS-rt locations feed more closely mirrors the RTR locations feed.

The discrepancies in what fields are provided in the GTFS-rt feed compared to the RTR feed necessitated some slight logic changes in the main predictions flow. Most notably, the `stops_away` field is not in the GTFS-rt field, but we are able to use locations data to mimic how `stops_away = 0` is determined as an alternative.

Changes can be tested locally against the GTFS-rt production feeds by using the following URLs for the `TRIP_UPDATE_URL` and `VEHICLE_POSITIONS_URL` env variables respectively
- `https://cdn.mbta.com/realtime/TripUpdates_enhanced.json`
- `https://cdn.mbta.com/realtime/VehiclePositions_enhanced.json`

Before this is merged/deployed, we will want to put up a Terraform PR to change the values of the aforementioned env variables first. Once that is merged and applied, then we can re-deploy RTS with the new logic.

PaulJKim commented on lib/predictions/predictions.ex:
```diff
@@ -10,11 +10,23 @@ defmodule Predictions.Predictions do
     predictions =
       feed_message["entity"]
       |> Enum.map(& &1["trip_update"])
+      |> Enum.filter(
```
Ah true. Might as well use the `Stream` versions of most of the rest of these chained operations too. Any opposition to that?

PaulJKim commented on lib/predictions/predictions.ex:
```diff
@@ -97,8 +108,9 @@ defmodule Predictions.Predictions do
-      stops_away: stop_time_update["stops_away"],
+      stopped?: not is_nil(vehicle_location) and vehicle_location.status == :stopped_at,
+      vehicle_at_predicted_stop?:
+        not is_nil(vehicle_location) and stop_time_update["stop_id"] == vehicle_location.stop_id,
```
Oh good call

panentheos commented on .envrc.template:
```diff
@@ -19,5 +19,5 @@ export API_V3_URL=https://api-dev-green.mbtace.com
 
 # URLs of the enhanced trip-update and vehicle-position feeds. Default to the real feed URLs if
 # not set here.
```
```suggestion
# URLs of the enhanced trip-update and vehicle-position feeds. Required.
```
We should update this comment.

panentheos reviewed (approved):
See one suggestion, then looks good! Tested by running locally.

PaulJKim commented:
@panentheos When if you have a minute, could I get another pass at this one?
Some minor changes since the last time
- Added the filtering to the last trip parsing logic
- Started filtering out past departures
- Cleaned up a couple references to RTR leftover

panentheos commented on lib/predictions/last_trip.ex:
```diff
@@ -2,7 +2,18 @@ defmodule Predictions.LastTrip do
+          "Green-D",
+          "Green-E",
+          "Mattapan"
+        ] and &1["trip"]["schedule_relationship"] != "CANCELED")
```
nit: We might want to split this route list into a helper, e.g. `relevant_rail_route?`, since we intend for both code paths to use the same list.

panentheos reviewed (approved):
See one comment, otherwise the changes look good.

PaulJKim commented:
Restoring the branch so we can deploy it to dev-green while we investigate what's going on with BRD predictions and passthrough announcements
